docs: Reword OTel Engine pages (#6781)
grafana/alloy: docs/sources/set-up/otel_engine.md

Service installers work seamlessly with the {{< param "OTEL_ENGINE" >}} as the feature matures.
In the meantime, use the CLI or Helm options for testing.

## Custom builds with the OpenTelemetry Collector Builder (OCB)

The {{< param "OTEL_ENGINE" >}} is generated from a declarative [OpenTelemetry Collector Builder (OCB)](https://opentelemetry.io/docs/collector/custom-collector/) manifest.
If you need additional components or want to remove bundled components, edit the manifest and build a customized {{< param "PRODUCT_NAME" >}} binary.

> Grafana doesn't offer commercial support for custom builds.

Author: It's a bit unusual for us to mention commercial support in a public doc. At the same time, I'd prefer to keep this sentence just to make sure customers don't expect us to support an OCB-built Alloy instance.

Reviewer: We do have precedent for this:

* https://grafana.com/docs/alloy/latest/get-started/components/community-components/
* https://github.com/grafana/alloy/blob/main/docs/sources/shared/stability/community.md

Service installers work seamlessly with the {{< param "OTEL_ENGINE" >}} as the feature matures.
In the meantime, use the CLI or Helm options for testing.

## Custom builds with the OpenTelemetry Collector Builder (OCB)

The {{< param "OTEL_ENGINE" >}} is generated from a declarative [OpenTelemetry Collector Builder (OCB)](https://opentelemetry.io/docs/collector/custom-collector/) manifest.
If you need additional components or want to remove bundled components, edit the manifest and build a customized {{< param "PRODUCT_NAME" >}} binary.
Grafana doesn't offer commercial support for custom builds.

> ### 1. Clone the {{< param "PRODUCT_NAME" >}} repository

Reviewer: Instead of making the steps into H3 headings, we could just use numbered steps and indent the body text 3 spaces.  This will group the step body correctly. 

If we keep the steps as H3s, we need to fix the variable callouts in the headings to use `%` instead of `< >`

Author: Sure! I changed it to numbers and it looks good.

>    Clone the Git repository and change to the repository root.

Reviewer: ```suggestion
1. Clone the {{< param "PRODUCT_NAME" >}} repository and change to the repository root.

```

We can collapse this and reduce redundancy.

Author: Ooof sorry, I wasn't thinking 🤦

   The following steps assume you run commands from this directory.

   ```shell
   git clone https://github.com/grafana/alloy.git
   cd alloy
   ```

   To build from a **specific release**, fetch tags and check out the tag after you clone:

   ```shell
   git fetch --tags
   git checkout <RELEASE_TAG>
   ```

   Replace _`<RELEASE_TAG>`_ with the [release tag](https://github.com/grafana/alloy/releases) you want.

> 1. Start from the checked-in manifest.

Reviewer: Start what? What should the user do in this step? Locate the manifest? edit it?

Author: Lol. This is how it's phrased [atm](https://grafana.com/docs/alloy/latest/introduction/otel_alloy/#2-start-from-the-checked-in-manifest). I'll reword it.

Author: Done

   The source manifest is [`collector/builder-config.yaml`](https://github.com/grafana/alloy/blob/main/collector/builder-config.yaml) in your checkout.
   You can:

   - **Remove** a component: delete its `- gomod: ...` line from the appropriate section.
   - **Add** a component: append a line that points at the module path and version you want.
     Follow the same `- gomod:` pattern as the other entries.

1. Build the {{< param "PRODUCT_NAME" >}} binary.

>    Build the full {{< param "PRODUCT_NAME" >}} binary:

Reviewer: ```suggestion
1. Build the full {{< param "PRODUCT_NAME" >}} binary.

```

   ```shell
   make alloy
   ```

   The binary in `build/` behaves like a standard `alloy` build.
   Use [`alloy otel`](../../reference/cli/otel/) to run collector YAML against your custom bundle.

1. Build a Docker image.

>    To create an image like the Grafana {{< param "PRODUCT_NAME" >}} image:

Reviewer: ```suggestion
1. Build the {{< param "PRODUCT_NAME" >}} Docker image.

```

Is this in addition to the binary build in the previous step? Or an optional "you can do it this way too' step?

Author: It's an optional step. It is independent of the previous one, because the binary which goes in the docker image is built separately.

Reviewer: We could word it so the user knows there's a choice and why it's separate?
